nodejs/FatBeaconPeripheral/FatBeacon.js

```diff
@@ -0,0 +1,60 @@
+var bleno = require('bleno');
+var eddystoneBeacon = require('eddystone-beacon');
+
+/*************************Altering Library Functions****************************/
+
+var AdvertisementData = require('eddystone-beacon/lib/util/advertisement-data');
+var Eir = require('eddystone-beacon/lib/util/eir');
+var FAT_BEACON_FRAME_TYPE = 0x0e;
+
+var makeUrlBufferReplacement = function(name) {
+  console.log('AdvertisementData.makeUrlBuffer Called with ' + name);
+
+  var FAT_BEACON_FRAME_TYPE = 0x0e;
+  var MAX_URL_LENGTH = 18;
+  var SERVICE_UUID = 'feaa';
+  
+  var encodedName = Buffer.from(name);
+
+  if (encodedName.length > MAX_URL_LENGTH) {
+    throw new Error('Encoded Name must be less than ' + MAX_URL_LENGTH +
+                    ' bytes. It is currently ' + encodedName.length + ' bytes.');
+  }
+
+  var data = Buffer.concat([
+    encodedName
+  ]);
+
+  var header = new Buffer(1);
+
+  header.writeUInt8(FAT_BEACON_FRAME_TYPE, 0);
+
+  var fatBeaconHeader = new Buffer(2);
+  fatBeaconHeader[0] = 0x10;
+  fatBeaconHeader[1] = 0xba;
+
+  var serviceData = Buffer.concat([
+    fatBeaconHeader,
+    header,
+    data
+  ]);
+
+  var eir = new Eir();
+
+  eir.addFlags(0x06);
+  eir.add16BitCompleteServiceList([SERVICE_UUID]);
+  eir.addServiceData(SERVICE_UUID, serviceData);
+
+  return eir.buffer();
+}
+
+var alterLibrariesForFatBeaconOptimization = function() {
+  console.log('Calling Optimize');
+  AdvertisementData.makeUrlBuffer = makeUrlBufferReplacement;
+}
+
+alterLibrariesForFatBeaconOptimization();
+
+/*********************************************************/
+
+eddystoneBeacon.advertiseUrl('New PhatBeacon');
```

Comment:
Does this work as-is?  What exactly does it do?  Does it show a beacon that can't actually be connected to?  Please flesh this out in the commit message.  Also, the line length limit on the commit message body is 72 (only the subject is 50).